### ISSUE TITLE
BBC programmes file & folder naming issues

### ISSUE THREAD

**Issue by approtech** [Ascoware/get-iplayer-automator#376]
*What steps will reproduce the problem? Be specific, and provide as much detail as possible.*

Download any BBC programme from iPlayer.

*What is the expected output? What do you see instead?*

Expected:
Folder name "The Sound of Petula Series 1"
File name "The Sound of Petula Series 1 - 06. Your Kind of Music ((hvfxsd)).mp4"

Actual:
Folder name "The Sound of Petula"
File name "The Sound of Petula Series 1 - 06. Your Kind of Music - b01pvc6y (hvfxsd).mp4"

*What version of GiA are you using? On what version of OS X?*

GiA v1.21.13, macOS v10.15.7
"Use Kodi (XBMC) naming" not ticked.

*Are you using a web proxy? DNS proxy? VPN? VPS?*

No to all.

*What program are you attempting to download (full title, series and episode)? **Provide only ONE program as an example.** *

Example programme used: "The Sound of Petula Series 1 Episode 6"
https://www.bbc.co.uk/iplayer/episode/b01pvc6y/the-sound-of-petula-series-1-6-your-kind-of-music

*Please provide any additional information below.*

No problem with the download, only the naming. "Use Kodi (XBMC) naming" not ticked. The 'expected' naming is exactly as generated by the previous version of GiA (v1.20.12), confirmed by downloading the same programme with that version.

So there are two distinct issues:
1) The folder name no longer includes the "Series n" reference
2) The file name now includes " - [programme Id" before the format and filetype

Note that using Kodi naming the programme ID is not included, but the folder still has no Series number. In some cases, not having the Series number in the folder name may be a benefit, but it should be a new option to include (the default, as previously) or exclude it (new in this version).

**Comment by skovatch** [comment]
Here's what I think happened. I made a change to directory and file naming to address a bug I found where two shows could end up with an identical name even though they had a different program ID. I was running with that locally for a while, and eventually deleted it, but then I built v1.21.13 which had that change in it. So it's in a build but not checked in.

I guess this is the motivation I need to have GitHub automate builds on commit. The next update won't have this issue.
